packages/components/src/slider/slider/hook.ts

```diff
@@ -0,0 +1,131 @@
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useMemo, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import * as styles from '../styles';
+import { useContextSystem, WordPressComponentProps } from '../../ui/context';
+import { useFormGroupContextId } from '../../ui/form-group';
+import { useControlledValue, useCx } from '../../utils/hooks';
+import { interpolate } from '../../utils/interpolate';
+import { parseCSSUnitValue, createCSSUnitValue } from '../../utils/unit-values';
+import { isValueNumeric } from '../../utils/values';
+
+import type { SliderProps } from '../types';
+
+const noop = () => {};
+
+export function useSlider(
+	props: WordPressComponentProps< SliderProps, 'input', false >
+) {
+	const {
+		className,
+		defaultValue,
+		error,
+		onBlur = noop,
+		onChange: onChangeProp = noop,
+		onFocus = noop,
+		id: idProp,
+		isFocused: isFocusedProp = false,
+		max = 100,
+		min = 0,
+		size = 'default',
+		style,
+		thumbColor,
+		trackColor,
+		trackBackgroundColor,
+		value: valueProp,
+		...otherProps
+	} = useContextSystem( props, 'Slider' );
+
+	const [ _value, onChange ] = useControlledValue( {
+		defaultValue: defaultValue || '50',
+		onChange: onChangeProp,
+		value: valueProp,
+	} );
+	const [ value, initialUnit ] = parseCSSUnitValue( `${ _value }` );
+
+	const id = useFormGroupContextId( idProp );
```

Comment:
The `FormGroup` readme say it's experimental at the moment. I'm not sure if it is ready for wider use but wanted to stay as close to the G2 version as possible using what was already available. Would this be better addressed in a follow-up?

Reply:
I gave [`FormControl`](https://github.com/WordPress/gutenberg/blob/e6e0ea8906c0fb01ea132e3386d5e42172614ba9/packages/components/src/ui/form-group/README.md) a deeper look — it seems like it partially overlaps with [`BaseControl`](https://github.com/WordPress/gutenberg/blob/c3a49add59883be4aadf5fda0014d97ac5ba79ee/packages/components/src/base-control/README.md).

I would probably switch to `BaseControl` for the time being, and later have a look at `FormGroup` and at the chance of consolidating `BaseControl` and `FormGroup` into one component (this may be also interesting / useful in the context of the recent challenge of labelling compound components that we're facing with `BorderBoxControl`)